src/fuchsia/mod.rs

```diff
@@ -1334,9 +1334,9 @@ pub const GRPQUOTA: ::c_int = 1;
 
 pub const SIGIOT: ::c_int = 6;
 
-pub const S_ISUID: ::c_int = 0x800;
-pub const S_ISGID: ::c_int = 0x400;
-pub const S_ISVTX: ::c_int = 0x200;
+pub const S_ISUID: ::mode_t = 0x800;
+pub const S_ISGID: ::mode_t = 0x400;
+pub const S_ISVTX: ::mode_t = 0x200;
```

Comment:
Not mandatory but this would eliminate an upcoming conflict with https://github.com/rust-lang/libc/pull/3634
```suggestion
pub const S_ISUID: ::mode_t = 0o4000;
pub const S_ISGID: ::mode_t = 0o2000;
pub const S_ISVTX: ::mode_t = 0o1000;
```